.ci/visual-regression-test.sh

```diff
@@ -0,0 +1,132 @@
+#!/bin/bash
+
+# Variables
+BUILD_DIR=$(pwd)
+GITHUB_API_URL="https://api.github.com/repos/$CI_PROJECT_USERNAME/$CI_PROJECT_REPONAME"
+
+# Check if we are NOT on the master branch and this is a PR
+if [[ ${CI_BRANCH} != "master" && -z ${CI_PR_URL} ]];
+then
+  echo -e "\nVisual regression tests will only run if we are not on the master branch and making a pull request"
+  exit 0;
+fi
+
+echo -e "\nProcessing pull request #$PR_NUMBER"
+
+GIT_FILE_MODIFIED()
+{
+    # Stash list of changed files
+    GIT_FILES_CHANGED="$(git diff master --name-only)"
+
+    while read -r changedFile; do
+        if [[ "${changedFile}" == "$1" ]]
+        then
+            return 0;
+        fi
+    done <<< "$GIT_FILES_CHANGED"
+
+    return 1;
+}
+
+LAST_GIT_COMMIT_MESSAGE=$(git log -1 --pretty=%B)
+
+# Always run visual tests if "[vr]" is in the last commit message
+if [[ ${LAST_GIT_COMMIT_MESSAGE} != *"[vr]"* ]]
+then
+
+    # Skip visual tests if there hasn't been a modification to composer.lock
+    if ! GIT_FILE_MODIFIED 'composer.lock'
+    then
+        echo -e "\nSkipping visual regression tests since composer.lock has NOT changed"
+        exit 0
+    fi
+
+    # Skip visual tests if has been a modification to composer.json
+    if GIT_FILE_MODIFIED 'composer.json'
+    then
+        echo -e "\nSkipping visual regression tests since composer.json HAS changed"
+        exit 0
+    fi
+
+else
+    echo -e "\nRunning visual regression tests because the latest commit message demands it"
+fi
+
+# Ping the multidev environment to wake it from sleep
+echo -e "\nPinging the ${TERMINUS_ENV} multidev environment to wake it from sleep..."
+curl -I "$MULTIDEV_SITE_URL" >/dev/null
+
+# Ping the live environment to wake it from sleep
+echo -e "\nPinging the live environment to wake it from sleep..."
+curl -I "$LIVE_SITE_URL" >/dev/null
+
+# Check for custom backstop.json
+if [ ! -f ./.ci/backstop.json ]; then
```

Comment:
So, I just had a thought about this.  If there is no backstop.json file, then the script will be run, and it will 
fill in scenario.url with `process.env.MULTIDEV_SITE_URL`. This value changes with every PR, correct?

It seems to me that if there is no backstop.json, then the default config file should be copied over to backstop.json, but thereafter, the contents of backstop.json should be replaced so that the dynamic values can be updated regardless of whether the user has provided customized values.